src/caret_analyze/common/type_check_decorator.py

```diff
@@ -185,26 +189,65 @@ def _get_given_arg_type(
             given_arg_type_str = f"'{given_arg[given_arg_loc[1]].__class__.__name__}'"
         elif error_type == 'IterableArg':
             given_arg_type_str = f"'{given_arg[int(given_arg_loc[1])].__class__.__name__}'"
-        else:
+        elif varargs is None:
             given_arg_type_str = f"'{given_arg.__class__.__name__}'"
+        else:
+            # For functions with variable length arguments,
+            given_arg_type_str = f"'{args[given_arg_loc[1]].__class__.__name__}'"
 
         return given_arg_type_str
 
+    def _parse_collection_or_unpack(
+        target_arg: tuple[Collection[Any]] | tuple[Any, ...]
+    ) -> tuple[Any]:
+        """
+        Parse target argument.
+
+        To address both cases where the target argument is passed in collection type
+        or unpacked, this function converts them to the same list format.
+
+        Parameters
+        ----------
+        target_arg : tuple[Collection[Any]] | tuple[Any, ...]
+            Target objects.
+
+        Returns
+        -------
+        tuple[Any]
+
+        """
+        parsed_target_objects: tuple[Any]
+        if isinstance(target_arg[0], Collection):
+            assert len(target_arg) == 1
+            parsed_target_objects = tuple(target_arg[0])
+        else:  # Unpacked case
+            parsed_target_objects = tuple(target_arg)  # type: ignore
+        return parsed_target_objects
+
     def decorator(func):
         validate_arguments_wrapper = \
             validate_arguments(config={'arbitrary_types_allowed': True})(func)
 
         @wraps(func)
         def _custom_wrapper(*args, **kwargs):
             try:
+                # Checks whether the arguments of a given func have variable length arguments
+                check_exist_varargs = inspect.getfullargspec(func)
```

Comment:
This variable is not appropriate.

`check_xxx` looks like it is an boolean value.
If this variable has other type, it should be named such that another programmer can know what kind of data it has.

Reply:
Fixed in https://github.com/tier4/caret_analyze/pull/472/commits/189c4949dc6b47fd9441d8198e464a7cc0c83b58.